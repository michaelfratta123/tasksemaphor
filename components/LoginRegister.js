// define a login/register component
import React, { useState } from "react";
import { Form, Button, Card } from "react-bootstrap";

const LoginRegister = ({ handleLogin, handleRegister }) => {
  const [formData, setFormData] = useState({
    username: "",
    password: "",
  });

  // save the inputted form data to the state
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData({
      ...formData,
      [name]: value,
    });
  };

  // call the relevant function depending on the button that was clicked (login/register)
  const handleAction = (action) => {
    if (action === "login") {
      handleLogin(formData);
    } else if (action === "register") {
      handleRegister(formData);
    }
  };

  // render the component
  return (
    <div>
      <Card
        style={{ width: "20rem" }}
        className="m-auto text-light"
        bg="dark"
        border="light"
      >
        <Card.Body>
          <Card.Title className="mb-4 mt-2">Login or Register</Card.Title>
          <Form>
            <Form.Group className="mb-3 mx-4" controlId="formBasicUsername">
              <Form.Control
                type="text"
                name="username"
                value={formData.username}
                onChange={handleChange}
                placeholder="Username"
                required
              />
            </Form.Group>
            <Form.Group className="mb-3 mx-4" controlId="formBasicPassword">
              <Form.Control
                type="password"
                name="password"
                value={formData.password}
                onChange={handleChange}
                placeholder="Password"
                required
              />
            </Form.Group>
            <Button
              className="m-3"
              variant="secondary"
              type="button"
              onClick={() => handleAction("login")}
            >
              Login
            </Button>
            <Button
              variant="secondary"
              type="button"
              onClick={() => handleAction("register")}
            >
              Register
            </Button>
          </Form>
        </Card.Body>
      </Card>
    </div>
  );
};

export default LoginRegister;
