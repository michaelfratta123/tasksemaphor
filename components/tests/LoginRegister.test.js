// Snapshot test for LoginRegister component
import React from "react";
import renderer from "react-test-renderer";
import LoginRegister from "../LoginRegister";

describe("LoginRegister Component Snapshot", () => {
  test("should match the snapshot", () => {
    const mockHandleLogin = jest.fn();
    const mockHandleRegister = jest.fn();

    const component = renderer.create(
      <LoginRegister
        handleLogin={mockHandleLogin}
        handleRegister={mockHandleRegister}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
