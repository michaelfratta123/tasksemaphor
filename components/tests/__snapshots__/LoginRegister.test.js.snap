// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginRegister Component Snapshot should match the snapshot 1`] = `
<div>
  <div
    className="m-auto text-light card bg-dark border-light"
    style={
      {
        "width": "20rem",
      }
    }
  >
    <div
      className="card-body"
    >
      <div
        className="mb-4 mt-2 card-title h5"
      >
        Login or Register
      </div>
      <form
        className=""
      >
        <div
          className="mb-3 mx-4"
        >
          <input
            className="form-control"
            id="formBasicUsername"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="mb-3 mx-4"
        >
          <input
            className="form-control"
            id="formBasicPassword"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <button
          className="m-3 btn btn-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
        <button
          className="btn btn-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Register
        </button>
      </form>
    </div>
  </div>
</div>
`;
